ios/brave-ios/Sources/Brave/Helpers/URLBarHelper.swift

```diff
@@ -0,0 +1,183 @@
+// Copyright 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import Foundation
+import UIKit
+
+class URLBarHelper {
+
+  static let shared = URLBarHelper()
+
+  func shouldShowSearchSuggestions(using lastReplacement: String) -> Bool {
+    // Check if last entry to url textfield needs to be checked as suspicious.
+    // The reason of checking count is bigger than 1 is the single character
+    // entries will always be safe and only way to achieve multi character entry is
+    // using paste board.
+    // This check also allow us to handle paste permission case
+    guard lastReplacement.count > 1 else {
+      return true
+    }
+
+    // Check if paste board has any text to guarantee the case
+    guard UIPasteboard.general.hasStrings || UIPasteboard.general.hasURLs else {
+      return true
+    }
+
+    // Perform check on pasted text
+    if let pasteboardContents = UIPasteboard.general.string {
+      return !isSuspiciousQuery(pasteboardContents)
+    }
+
+    return true
+  }
+
+  /// Whether  the desired text should allow search suggestions to appear when it is copied
+  /// - Parameter query: Search query copied
+  /// - Returns: the result if it is suspicious
+  func isSuspiciousQuery(_ query: String) -> Bool {
```

Comment:
Have you done any performance testing on this method? Does it need to be done off main perhaps?

Reply:
Yes it is quite fast, like instantaneous thats why I didn't even eadd performance test but this is great idea.

I dont see any reason not to do it since our case is covering pasting, will add.